exe/facter.cc

```diff
@@ -264,6 +264,10 @@ int main(int argc, char **argv)
         bool show_legacy = vm.count("show-legacy");
         facts.write(boost::nowide::cout, fmt, queries, show_legacy);
         boost::nowide::cout << endl;
+
+        if (ruby) {
```

Comment:
we need to use `leatherman::util::scope_exit()` when we initialize ruby isntead of doing a conditional cleanup here, to protect against exceptions.